#!/usr/bin/env python
PACKAGE = "franka_example_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("translational_stiffness", double_t, 0, "Cartesian translational stiffness", 200, 0, 400)
gen.add("rotational_stiffness", double_t, 0, "Cartesian rotational stiffness", 10, 0, 30)
# gen.add("nullspace_stiffness", double_t, 0, "Stiffness of the joint space nullspace controller (the desired configuration is the one at startup)", 0.5, 0, 100)
gen.add("nullspace_stiffness", double_t, 0, "Stiffness of the joint space nullspace controller (the desired configuration is the one at startup)", 0.05, 0, 100)
gen.add("admittance_translational_stiffness", double_t, 0, "admittance_translational_stiffness", 100, 0, 400)
gen.add("admittance_rotational_stiffness", double_t, 0, "admittance_translational_stiffness", 50, 0, 40)
gen.add("admittance_translational_damping", double_t, 0, "admittance_translational_damping", 0, 10, 80)
gen.add("admittance_rotational_damping", double_t, 0, "admittance_translational_damping", 0, 3, 10)
gen.add("admittance_translational_mass", double_t, 0, "admittance_translational_mass", 0, 0.01, 1)
gen.add("admittance_rotational_mass", double_t, 0, "admittance_translational_mass", 0, 0.001, 1)

exit(gen.generate(PACKAGE, "dynamic_compliance", "compliance_param"))
